tests/integration.rs

```diff
@@ -0,0 +1,433 @@
+
+        let req = Request::builder()
+            .method(Method::GET)
+            .uri(format!("{}/roots/0", self.pebble_management_url()))
+            .header(CONTENT_TYPE, "application/json")
+            .body(Full::default())?;
+
+        let resp = client.request(req).await?;
+        if resp.status() != 200 {
+            return Err(format!("unexpected /roots/0 response status: {}", resp.status()).into());
+        }
+
+        let body = resp.collect().await?.aggregate();
+        let mut pem = String::new();
+        body.reader().read_to_string(&mut pem)?;
+        let root = CertificateDer::from_pem_slice(pem.as_bytes())?;
+
+        let mut roots = RootCertStore::empty();
+        roots.add_parsable_certificates(vec![root]);
+        assert_eq!(roots.len(), 1);
+        Ok(roots)
+    }
+
+    fn challenge_management_url(&self) -> &str {
+        "http://127.0.0.1:8055" // Default.
+    }
+
+    fn pebble_management_url(&self) -> String {
+        format!("https://{}", &self.config.pebble.management_listen_address)
+    }
+
+    fn directory_url(&self) -> String {
+        format!("https://{}/dir", &self.config.pebble.listen_address)
+    }
+}
+
+#[derive(Clone, Serialize)]
+struct Config {
+    pebble: PebbleConfig,
+}
+
+#[derive(Clone, Serialize)]
+struct PebbleConfig {
+    #[serde(rename = "listenAddress")]
+    listen_address: String,
+    #[serde(rename = "managementListenAddress")]
+    management_listen_address: String,
+    certificate: String,
+    #[serde(rename = "privateKey")]
+    private_key: String,
+    #[serde(rename = "httpPort")]
+    http_port: u16,
+    #[serde(rename = "tlsPort")]
+    tls_port: u16,
+    #[serde(rename = "ocspResponderURL")]
+    ocsp_responder_url: String,
+    #[serde(rename = "externalAccountBindingRequired")]
+    external_account_binding_required: bool,
+    #[serde(rename = "domainBlocklist")]
+    domain_blocklist: Vec<String>,
+    retry_after: RetryConfig,
+    profiles: HashMap<String, Profile>,
+}
+
+#[derive(Clone, Serialize)]
+struct RetryConfig {
+    authz: u32,
+    order: u32,
+}
+
+#[derive(Clone, Serialize)]
+struct Profile {
+    description: String,
+    #[serde(rename = "validityPeriod")]
+    validity_period: u32,
+}
+
+struct Subprocess(Option<Child>);
+
+impl Subprocess {
+    fn new(cmd: &mut Command) -> io::Result<Self> {
+        Ok(Self(Some(cmd.spawn()?)))
+    }
+}
+
+impl Drop for Subprocess {
+    fn drop(&mut self) {
+        if let Some(mut child) = self.0.take() {
+            child.kill().expect("failed to kill subprocess");
+            child.wait().expect("failed to wait for killed subprocess");
+        }
+    }
+}
+
+fn wait_for_server(addr: &str) {
+    for i in 0..10 {
+        if TcpStream::connect(addr).is_ok() {
+            return;
+        }
+        thread::sleep(Duration::from_millis(i * 100));
+    }
+    panic!("failed to connect to {:?} after 10 tries", addr);
+}
+
+static DEFAULT_CONFIG: LazyLock<Config> = LazyLock::new(|| Config {
```

Comment:
The use of `LazyLock` means this can't work with the current MSRV, I think? That might be okay I guess to assume we'll only run this on stable?

Reply:
>  That might be okay I guess to assume we'll only run this on stable?

That was my thinking (and what the CI addition does), but I think it will fall away if I switch to the static lifetimes as suggested in the other comment. Will circle back depending on how that goes.

Reply:
This fell away.